design/metastore.md

```diff
@@ -0,0 +1,102 @@
+# Next Generation Metastore
+
+### Overview
+
+Deliver manageability of metastore information alongside the object-store data as part of lakeFS.
+
+### Goals
+
+- Metastore entities are versioned: using the same commit and branching abilities that lakeFS provides
+- Metastore entities are versioned together with the data: an atomic merge operation should contain both data mutations as well as metastore mutations.
+- Metastore entities are diff-able: can view the changes made to the metadata between commits
+- Maintain some compatibility with existing HMS, or at least a narrow set of its versions and API
+- Extend functionality to also support next-generation table formats in addition to hive-style tables (Notably, Iceberg and Delta Lake) - long term goal.
+
+
+### Non-Goals
+
+- Support storing metadata inside existing Hive Metastore implementations
+- Provide any significant change in performance or throughput of metadata operations
+- Provide compatibility with other specific versions of Hive or Hive Metastore
+
+
+### Design Diagram
+
+[lakeFS Metastore](diagrams/metastore.png)
+
+In this diagram:
```

Comment:
Can we direct to the backing S3 store implementation?  This will make some things much easier and faster.